org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/ChangeUtil.java

```diff
@@ -116,8 +112,18 @@ public static void convertChanges(Change change, CheWorkspaceEdit edit) throws C
     return result;
   }
 
-  private static void convertCompositeChange(Change change, CheWorkspaceEdit edit)
-      throws CoreException {
+  private static void convertCompositeChange(
+      Change change, CheWorkspaceEdit edit, IProgressMonitor pm) throws CoreException {
+    if (change instanceof CompositeChange) {
+      for (Change child : ((CompositeChange) change).getChildren()) {
+        convertCompositeChange(child, edit, pm);
+      }
+    }
+    if (change instanceof TextFileChange) {
+      convertTextFileChange((TextFileChange) change, edit, pm);
+      return;
+    }
+
```

Comment:
Btw: if modifiedElement is null, it's not instanceof IJavaElement. The null check can go

Reply:
Does a ResourceChange have a modified element and does it have to be a IJavaElement? If not, the modifiedElement stuff needs to go inside the "if (change instanceof TextChange)".

Reply:
fixed